Add operators that are missing on the detail page (#839)
neo4j/docs-cypher: modules/ROOT/pages/planning-and-tuning/operators/operators-detail.adoc

======

[[query-plan-directed-relationship-by-element-id-seek]]
== Directed Relationship By Element Id Seek
// DirectedRelationshipByElementIdSeek

> The `DirectedRelationshipByElementIdSeek` operator reads one or more relationships by element id from the relationship store, and produces both the relationship and the nodes on either side.

Reviewer: What is meant by "on either side" here? Do you mean "and the nodes on either side of the relationship" or something else?

```suggestion
The `DirectedRelationshipByElementIdSeek` operator reads one or more relationships by element id from the relationship store (specified via the function xref::functions/scalar.adoc#functions-elementid[elementId()]) and produces both the relationship and the nodes on either side.
```

+-------------------------------+----+---------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+
| +ProduceResults               |  0 | r, n1                                       |              1 |    1 |       7 |              0 |                        |           |                     |
| |                             +----+---------------------------------------------+----------------+------+---------+----------------+                        |           |                     |
| +DirectedRelationshipByIdSeek |  1 | (n1)-[r]->(anon_0) WHERE id(r) = $autoint_0 |              1 |    1 |       1 |            248 |                    3/0 |     0.483 | Fused in Pipeline 0 |
+-------------------------------+----+---------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+

Total database accesses: 8, total allocated memory: 312
----

======

[[query-plan-undirected-relationship-by-element-id-seek]]
== Undirected Relationship By Element Id Seek
// UndirectedRelationshipByElementIdSeek

> The `UndirectedRelationshipByElementIdSeek` operator reads one or more relationships by element id from the relationship store.

Reviewer: ```suggestion
The `UndirectedRelationshipByElementIdSeek` operator reads one or more relationships by element id from the relationship store (specified via the function xref::functions/scalar.adoc#functions-elementid[elementId()]).
```

======

[[query-plan-undirected-relationship-by-id-seek]]
== Undirected Relationship By Id Seek
// UndirectedRelationshipByIdSeek

> The `UndirectedRelationshipByIdSeek` operator reads one or more relationships by id from the relationship store.

Reviewer: ```suggestion
The `UndirectedRelationshipByIdSeek` operator reads one or more relationships by id from the relationship store (specified via the function xref::functions/scalar.adoc#functions-elementid[elementId()]).
```

== Node By Id Seek
// NodeByIdSeek

> The `NodeByIdSeek` operator reads one or more nodes by ID from the node store, specified via the function xref::functions/scalar.adoc#functions-id[id()].

Reviewer: ```suggestion
The `NodeByIdSeek` operator reads one or more nodes by id from the node store, specified via the function xref::functions/scalar.adoc#functions-id[id()].
```

| Operator        | Id | Details                    | Estimated Rows | Rows | DB Hits | Memory (Bytes) | Page Cache Hits/Misses | Time (ms) | Pipeline            |
+-----------------+----+----------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+
| +ProduceResults |  0 | n                          |              1 |    1 |       2 |              0 |                        |           |                     |
| |               +----+----------------------------+----------------+------+---------+----------------+                        |           |                     |
| +NodeByIdSeek   |  1 | n WHERE id(n) = $autoint_0 |              1 |    1 |       1 |            248 |                    2/0 |     1.109 | Fused in Pipeline 0 |
+-----------------+----+----------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+

Total database accesses: 3, total allocated memory: 312

1 row
----

======

[[query-plan-node-by-element-id-seek]]
== Node By Element Id Seek
// NodeByElementIdSeek

> The `NodeByElementIdSeek` operator reads one or more nodes by element ID from the node store, specified via the function xref::functions/scalar.adoc#functions-elementid[elementId()].

Reviewer: ```suggestion
The `NodeByElementIdSeek` operator reads one or more nodes by element id from the node store, specified via the function xref::functions/scalar.adoc#functions-elementid[elementId()].
```

======

[query-plan-transaction-apply]]
== TransactionApply
// TransactionApply

> `TransactionApply` works like normal `Apply` but will commit the current transaction after a specified number of rows.

Reviewer: ```suggestion
`TransactionApply` works like the `Apply` operator but will commit the current transaction after a specified number of rows.
```

======

[[query-plan-nullify-metadata]]
== Nullify Metadata
// NullifyMetadata

> _This feature was introduced in Neo4j 5.9._
> NullifyMetadata is responsible for cleaning up state associated with Repeat(Trail), it is only planned directly after Repeat(Trail).

Reviewer: Im not sure if my suggestion to add a "the" before "state" is correct here. In any case, can we explain briefly what the "state" refers to here?

```suggestion
_This feature was introduced in Neo4j 5.9._

`NullifyMetadata` is responsible for cleaning up the state associated with `Repeat(Trail)`.
It is only planned directly after `Repeat(Trail)`.
```

======

[query-plan-transaction-foreach]]
== TransactionForeach
// TransactionForeach

> `TransactionForeach` works like normal `Foreach` but will commit the current transaction after a specified number of rows.

Reviewer: ```suggestion
`TransactionForeach` works like the `Foreach` operator but will commit the current transaction after a specified number of rows.
```

 | | |                 +----+--------------------------------------------------+----------------+------+---------+----------------+                        |           |                     |
 | | +Argument         |  4 | line                                             |             10 |    4 |       0 |           3472 |                    0/0 |     0.712 | Fused in Pipeline 2 |
 | |                   +----+--------------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+
 | +LoadCSV            |  5 | line                                             |             10 |    4 |       0 |            328 |                        |           | In Pipeline 1       |
 +---------------------+----+--------------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+

 Total database accesses: 12, total allocated memory: 5704
----

======

[query-plan-subquery-foreach]]
== SubqueryForeach
// SubqueryForeach

> `SubqueryForeach` like normal `Foreach` but used for executing sub-queries.

Reviewer: ```suggestion
`SubqueryForeach` works like the `Foreach` operator but it is only used for executing subqueries.
```

======

[[query-plan-argument-tracker]]
== Argument Tracker
// ArgumentTracker

> The `ArgumentTracker` operator is used to make sure we have row-by-row semantics.
> This restricts the runtime not to batch operations in larger chunks.

Reviewer: ```suggestion
The `ArgumentTracker` operator is used to ensure row-by-row semantics.
This restricts the xref:planning-and-tuning/runtimes/index.adoc[Cypher runtime] to not batch operations in larger chunks.
```

| | +SetProperty     |  4 | s.seen = coalesce(s.seen + $autoint_1, $autoint_2)    |              0 |    0 |       0 |                |                        |           |                     |
| | |                +----+-------------------------------------------------------+----------------+------+---------+----------------+                        |           |                     |
| | +Argument        |  5 | s                                                     |              0 |    0 |       0 |            256 |                    0/0 |     0.000 | Fused in Pipeline 1 |
| |                  +----+-------------------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+
| +NodeIndexSeek     |  6 | RANGE INDEX s:Person(name) WHERE name = $autostring_0 |              0 |    0 |       1 |            376 |                    1/0 |     0.286 | In Pipeline 0       |
+--------------------+----+-------------------------------------------------------+----------------+------+---------+----------------+------------------------+-----------+---------------------+

> Total database accesses: 1, total allocated memory: 2104

Reviewer: ```suggestion
```
